File: cash.c
#include <cs50.h>
#include<stdio.h>
#include<math.h>

int main(void)
{
    float dollar;
    int cents, coins; // all variabels
    do
    {
        dollar = get_float("enter your change:"); // users change input
    }
    while (dollar <= 0);
    
    cents = round(dollar * 100);
    coins = 0; //variabels assignment 
    
    while (cents >= 25) //25 cents
    {
        cents -= 25;
        coins++;
    }
    while (cents >= 10) //10 cents
    {
        
        cents -= 10;
        coins++;
    }
    while (cents >= 5) //5 cents
    {
        
        cents -= 5;
        coins++;
    }
    while (cents >= 1) // 1 cent
    {
        
        cents -= 1;
        coins++;
    }

    printf("you will need at least %i coins\n", coins); //coins output  
    
}

File: cash.py
# fav libary
from cs50 import get_float 
# user input
while True:
    dn = get_float("change owed: ")
    if dn > 0 or dn == 0:
        break
cen = round(dn * 100)  # num of cents in input
coins = 0  # coins needed
while cen >= 25:  # quarter
    coins += 1
    cen -= 25
    
while cen >= 10:  # dime
    coins += 1
    cen -= 10
    
while cen >= 5:  # 5 cents
    coins += 1
    cen -= 5

while cen >= 1:  # penny
    coins += 1
    cen -= 1
    
print(coins)    